src/common/stringutils.cpp

```diff
@@ -172,21 +178,24 @@ std::wstring StrUtils::Utf8StringToUnicode(const std::string &str)
 int StrUtils::Utf8CharSizeAt(const std::string &str, unsigned int pos)
 {
     if (pos >= str.size())
-        return 0;
+        throw std::out_of_range("Index is greater than size");
```

Comment:
If that exception is needed, you might as well use `str.at(pos)` instead of `[]` and get it given to you without a prior check: https://en.cppreference.com/w/cpp/string/basic_string/at